### PR TITLE
Fix docs for memory-swap

### PR DESCRIPTION
**Pull request by hqhq**
Fixes: #18894

Signed-off-by: Qiang Huang h.huangqiang@huawei.com

Carry #18902 
ping @thaJeztah @moxiegirl

### CODE REVIEW

File: api/client/build.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	pull := cmd.Bool([]string{"-pull"}, false, "Always attempt to pull a newer version of the image")
 	dockerfileName := cmd.String([]string{"f", "-file"}, "", "Name of the Dockerfile (Default is 'PATH/Dockerfile')")
 	flMemoryString := cmd.String([]string{"m", "-memory"}, "", "Memory limit")
-	flMemorySwap := cmd.String([]string{"-memory-swap"}, "", "Total memory (memory + swap), '-1' to disable swap")
+	flMemorySwap := cmd.String([]string{"-memory-swap"}, "", "Swap limit equal to memory plus swap: '-1' to enable ulimited")
```
Comment (thaJeztah) [inline api/client/build.go]: typo: s/ulimited/unlimited/

File: docs/reference/commandline/build.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ parent = "smn_cli"
       --help                          Print usage
       --isolation=""                  Container isolation technology
       -m, --memory=""                 Memory limit for all build containers
-      --memory-swap=""                Total memory (memory + swap), `-1` to disable swap
+      --memory-swap=""                An positive integer equal to memory plus swap. Specify -1 to enable unlimited swap.
```
Comment (thaJeztah) [inline docs/reference/commandline/build.md]: (Not your fault); s/An positive/A positive/

File: docs/reference/commandline/create.md [hunk 3]
```diff
@@ -57,7 +57,7 @@ Creates a new container.
       -m, --memory=""               Memory limit
       --mac-address=""              Container MAC address (e.g. 92:d0:c6:0a:29:33)
       --memory-reservation=""       Memory soft limit
-      --memory-swap=""              Total memory (memory + swap), '-1' to disable swap
+      --memory-swap=""              An positive integer equal to memory plus swap. Specify -1 to enable unlimited swap.
```
Comment (thaJeztah) [inline docs/reference/commandline/create.md]: Same here; s/An positive/A positive/

File: docs/reference/commandline/run.md [hunk 4]
```diff
@@ -56,7 +56,7 @@ parent = "smn_cli"
       -m, --memory=""               Memory limit
       --mac-address=""              Container MAC address (e.g. 92:d0:c6:0a:29:33)
       --memory-reservation=""       Memory soft limit
-      --memory-swap=""              Total memory (memory + swap), '-1' to disable swap
+      --memory-swap=""              An positive integer equal to memory plus swap. Specify -1 to enable unlimited swap.
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md]: and here; s/An positive/A positive/

File: man/docker-build.1.md [hunk 5]
```diff
@@ -92,8 +92,14 @@ set as the **URL**, the repository is cloned locally and then sent as the contex
 **-m**, **--memory**=*MEMORY*
   Memory limit
 
-**--memory-swap**=*MEMORY-SWAP*
-  Total memory (memory + swap), '-1' to disable swap.
+**--memory-swap**=*LIMIT*
+   A limit value equal to memory plus swap. Must be used with the  **-m**
+(**--memory**) flag. This value should always larger than **-m**(**--memory**)
```
Comment (thaJeztah) [inline man/docker-build.1.md]: should always -> should always be

File: man/docker-build.1.md [hunk 6]
```diff
@@ -92,8 +92,14 @@ set as the **URL**, the repository is cloned locally and then sent as the contex
...
+value.
+
+   The format of `LIMIT` is <`number`>[<`unit`>]. Unit can be `b` (bytes),
```
Comment (thaJeztah) [inline man/docker-build.1.md]: can you put the back-tics around the whole "format" here?

    The format of `LIMIT` is `<number>[<unit>]`. Unit can be `b` (bytes),

File: man/docker-create.1.md [hunk 7]
```diff
@@ -230,11 +230,14 @@ reservation. So you should always set the value below **--memory**, otherwise th
 hard limit will take precedence. By default, memory reservation will be the same
 as memory limit.
 
-**--memory-swap**=""
-   Total memory limit (memory + swap)
+**--memory-swap**="LIMIT"
+   A limit value equal to memory plus swap. Must be used with the  **-m**
+(**--memory**) flag. This value should always larger than **-m**(**--memory**)
```
Comment (thaJeztah) [inline man/docker-create.1.md]: always -> always be

File: man/docker-create.1.md [hunk 8]
```diff
@@ -230,11 +230,14 @@ reservation. So you should always set the value below **--memory**, otherwise th
...
+value.
 
-   Set `-1` to disable swap (format: <number>[<unit>], where unit = b, k, m or g).
-This value should always larger than **-m**, so you should always use this with **-m**.
+   The format of `LIMIT` is <`number`>[<`unit`>]. Unit can be `b` (bytes),
```
Comment (thaJeztah) [inline man/docker-create.1.md]: same here with tics;

    The format of `LIMIT` is `<number>[<unit>]`. Unit can be `b` (bytes),

File: man/docker-run.1.md [hunk 9]
```diff
@@ -327,11 +327,14 @@ reservation. So you should always set the value below **--memory**, otherwise th
...
-**--memory-swap**=""
-   Total memory limit (memory + swap)
+**--memory-swap**="LIMIT"
+   A limit value equal to memory plus swap. Must be used with the  **-m**
+(**--memory**) flag. This value should always larger than **-m**(**--memory**)
+value.
 
-   Set `-1` to disable swap (format: <number>[<unit>], where unit = b, k, m or g).
-This value should always larger than **-m**, so you should always use this with **-m**.
+   The format of `LIMIT` is <`number`>[<`unit`>]. Unit can be `b` (bytes),
```
Comment (thaJeztah) [inline man/docker-run.1.md]: same here

Comment (thaJeztah) [inline man/docker-run.1.md]: hm, I just noticed @moxiegirl provided this in her example; @moxiegirl did you do so with a specific reason? It differs from the other places where we show a "format"

File: runconfig/parse.go [hunk 10]
```diff
@@ -93,7 +93,7 @@ func Parse(cmd *flag.FlagSet, args []string) (*container.Config, *container.Host
 		flHostname          = cmd.String([]string{"h", "-hostname"}, "", "Container host name")
 		flMemoryString      = cmd.String([]string{"m", "-memory"}, "", "Memory limit")
 		flMemoryReservation = cmd.String([]string{"-memory-reservation"}, "", "Memory soft limit")
-		flMemorySwap        = cmd.String([]string{"-memory-swap"}, "", "Total memory (memory + swap), '-1' to disable swap")
+		flMemorySwap        = cmd.String([]string{"-memory-swap"}, "", "Swap limit equal to memory plus swap: '-1' to enable ulimited")
```
Comment (thaJeztah) [inline runconfig/parse.go]: typo: s/ulimited/unlimited/

**Comment by hqhq** [comment]
@thaJeztah Thanks, updated.

**Comment by thaJeztah** [comment]
Thanks! LGTM

ping @moxiegirl for review :)

File: api/client/build.go [hunk 11]
```diff
@@ -49,7 +49,7 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	pull := cmd.Bool([]string{"-pull"}, false, "Always attempt to pull a newer version of the image")
 	dockerfileName := cmd.String([]string{"f", "-file"}, "", "Name of the Dockerfile (Default is 'PATH/Dockerfile')")
 	flMemoryString := cmd.String([]string{"m", "-memory"}, "", "Memory limit")
-	flMemorySwap := cmd.String([]string{"-memory-swap"}, "", "Total memory (memory + swap), '-1' to disable swap")
+	flMemorySwap := cmd.String([]string{"-memory-swap"}, "", "Swap limit equal to memory plus swap: '-1' to enable unlimited")
```
Comment (moxiegirl) [inline api/client/build.go]: @hqhq I'm sorry for this nit can you add "swap" after unlimited "enable unlimited swap"

File: man/docker-build.1.md [hunk 12]
```diff
@@ -92,8 +92,14 @@ set as the **URL**, the repository is cloned locally and then sent as the contex
 **-m**, **--memory**=*MEMORY*
   Memory limit
 
-**--memory-swap**=*MEMORY-SWAP*
-  Total memory (memory + swap), '-1' to disable swap.
+**--memory-swap**=*LIMIT*
+   A limit value equal to memory plus swap. Must be used with the  **-m**
+(**--memory**) flag. This value should always be larger than **-m**(**--memory**)
```
Comment (moxiegirl) [inline man/docker-build.1.md]: "This value" --> The swap `LIMIT` 

here an other instances.

File: runconfig/parse.go [hunk 13]
```diff
@@ -93,7 +93,7 @@ func Parse(cmd *flag.FlagSet, args []string) (*container.Config, *container.Host
 		flHostname          = cmd.String([]string{"h", "-hostname"}, "", "Container host name")
 		flMemoryString      = cmd.String([]string{"m", "-memory"}, "", "Memory limit")
 		flMemoryReservation = cmd.String([]string{"-memory-reservation"}, "", "Memory soft limit")
-		flMemorySwap        = cmd.String([]string{"-memory-swap"}, "", "Total memory (memory + swap), '-1' to disable swap")
+		flMemorySwap        = cmd.String([]string{"-memory-swap"}, "", "Swap limit equal to memory plus swap: '-1' to enable unlimited")
```
Comment (moxiegirl) [inline runconfig/parse.go]: enable unlimited -> enable  unlimited swap

**Comment by moxiegirl** [comment]
@hqhq a few last minute nits to fix. I apologize for the fragmented review.

**Comment by hqhq** [comment]
@moxiegirl Updated, thanks.

**Comment by thaJeztah** [comment]
still LGTM, thanks!

**Comment by moxiegirl** [comment]
LGTM @hqhq Thank you for the work!

**Pull request merged**